lib/spark.js

```diff
@@ -220,6 +239,37 @@ export default class Spark {
     }
   }
 
+  async checkRetrievalFromAlternativeProvider({
+    alternativeProviders,
+    randomness,
+    cid,
+  }) {
+    if (!alternativeProviders.length) {
+      console.info('No alternative providers found for this CID.')
+      return
+    }
+
+    const randomProvider = pickRandomProvider(alternativeProviders, randomness)
+    if (!randomProvider) {
+      console.warn(
+        'No providers serving the content via HTTP or Graphsync found. Skipping network-wide retrieval check.',
+      )
+      return
+    }
```

Comment:
Can we prevent this case earlier, when we decide whether to run this function in the first place?

Reply:
Yes, It's possible to prevent it by filtering alternative providers by their protocol. If we only have alternative providers that are serving content via `bitswap` by filtering them out we can exit early.